### DISCUSSION TITLE
Restore a Universal Artifact with MSBuild

### DISCUSSION THREAD

**Discussion by stan-sz** [dotnet/msbuild#7033] [Q&A] [answered]
Is there a way to restore a [universal package](https://docs.microsoft.com/en-us/azure/devops/artifacts/quickstarts/universal-packages?view=azure-devops) using MSBuild?

**Comment by rainersigwald** [comment, marked as answer]
Nothing built into MSBuild. You might be able to use [DownloadFile](https://docs.microsoft.com/visualstudio/msbuild/downloadfile-task), but I suspect auth would be a problem.

Easiest way is probably to `Exec` the `az artifacts universal download` command line. It's also possible that someone has written a task that uses the API to download, but I'm not aware of it if so.
